sqlmesh/core/schema_diff.py

```diff
@@ -453,7 +453,9 @@ def _alter_operation(
         root_struct: exp.DataType,
         new_kwarg: exp.ColumnDef,
     ) -> t.List[TableAlterOperation]:
-        current_type = exp.DataType.build(current_type)
+        # We don't copy on purpose here because current_type may need to be mutated inside
```

Comment:
why does data.build mutate?

Reply:
It doesn't mutate, it copies `current_type` which caused an issue in this module because we are mutating the struct expression in several places (search for `.pop` and `.insert` calls, as mentioned in the comment)

Reply:
so before my sqlglot commit we weren't copying inside `DataType.build` and this worked because the mutations were being propagated here, now that we copy they don't unless I switch it off here